### DISCUSSION TITLE
Operator overloading in C++

### DISCUSSION THREAD

**Discussion by benruijl** [rust-diplomat/diplomat#481] [General]
Is there a way to implement `operator+` and company for a `diplomat::opaque` type directly from Rust? If not, how would you recommend that I patch the auto-generated C++ files?

**Comment by Manishearth** [comment]
Not at the moment. We could potentially add this in the long run.

I would suggest writing a script that patches. We do also want the ability to inject arbitrary code into class definitions in a cleaner way, but we don't have that now.
